runtime/src/hardened_unpack.rs

```diff
@@ -167,24 +166,46 @@ pub fn unpack_snapshot<A: Read, P: AsRef<Path>>(
     )
 }
 
-fn is_valid_snapshot_archive_entry(parts: &[&str], kind: tar::EntryType) -> bool {
-    let like_storage = Regex::new(r"^\d+\.\d+$").unwrap();
-    let like_slot = Regex::new(r"^\d+$").unwrap();
```

Comment:
maybe, these regex isn't jit compiled? unfortunate. xD

Reply:
Can't say I've ever come across a regex that is truly fast, but maybe :)